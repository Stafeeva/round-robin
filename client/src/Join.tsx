import React from "react";

function Join({ joinMeeting }: { joinMeeting: (name: string) => void }) {
  const [name, setName] = React.useState<string>("");

  const handleJoin = async () => {
    await fetch("/api/join", {
      method: "POST",
      body: JSON.stringify({ name }),
      headers: {
        "Content-Type": "application/json",
      },
    });

    joinMeeting(name);
  };

  return (
    <div className="App">
      <main>
        <div className="container">
          <div className="join">
            <h2>Join</h2>
            <p>Please enter your name:</p>
            <form
              onSubmit={(e) => {
                e.preventDefault();
                handleJoin();
              }}
            >
              <input onChange={(e) => setName(e.currentTarget.value)} />
              <button type="submit">Join</button>
            </form>
          </div>
        </div>
      </main>
    </div>
  );
}

export default Join;
