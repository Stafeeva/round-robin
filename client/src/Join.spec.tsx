import React from "react";
import { render } from "@testing-library/react";

import Join from "./Join";
import userEvent from "@testing-library/user-event";

test("Join component renders", () => {
  const { container } = render(<Join joinMeeting={() => null} />);
  expect(container).toMatchSnapshot();
});

test("User can input their name", () => {
  const { getByText } = render(<Join joinMeeting={() => null} />);
  const input = getByText("Please enter your name:");
  expect(input).toBeInTheDocument();
});

// test("User can join a meeting", () => {
//   const { getByText } = render(<Join joinMeeting={() => null} />);
//   const input = getByText("Join");
//   userEvent.click(input);
//   expect(input).toBeInTheDocument();
// });
