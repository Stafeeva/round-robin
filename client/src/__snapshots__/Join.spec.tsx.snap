// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Join component renders 1`] = `
<div>
  <div
    class="App"
  >
    <main>
      <div
        class="container"
      >
        <div
          class="join"
        >
          <h2>
            Join
          </h2>
          <p>
            Please enter your name:
          </p>
          <form>
            <input />
            <button
              type="submit"
            >
              Join
            </button>
          </form>
        </div>
      </div>
    </main>
  </div>
</div>
`;
